src/components/Card/Card.js

```diff
@@ -1,32 +1,75 @@
 import styles from "./card.module.css";
 import { removeHTMLTags } from "../../utils/utils";
-import { BsPin, BsFillPinFill } from "react-icons/bs";
+import {
+  BsPin,
+  BsFillPinFill,
+  BsTrashFill,
+  BsTrash,
+  BsArchive,
+  BsArchiveFill,
+} from "react-icons/bs";
+import ColorPalette from "./Colors/Colors";
+
 
 export const Card = ({
+  id,
   title,
   archived,
   body,
   createdAt,
   updatedAt,
   pinned,
   trashed,
+  tags,
+  color,
 }) => {
   body = removeHTMLTags(body);
   if (title.length > 20) {
     title = title.substr(0, 20) + "...";
   }
+
   if (body.length > 100) {
     body = title.substr(0, 50) + "...";
   }
+
+  let dateCreated = createdAt
+    .toDate()
+    .toString()
+    .split(" ")
+    .splice(0, 4)
+    .join(" ");
+
   return (
-    <div className={styles.card}>
+    <div
+      style={{ background: `${color ? `${color}` : "fff"} ` }}
+      className={styles.card}
+    >
       <section className={styles.cardBody}>
         <article>
           <h6>{title}</h6>
           <p>{body}</p>
         </article>
         <span className="icon">{pinned ? <BsFillPinFill /> : <BsPin />}</span>
       </section>
+      <section className={styles.tags}>
+        {tags.map((tag, index) => {
+          return (
+            <p key={index} className={styles.tag}>
```

Comment:
one suggestion would be to replace `index` with an `id` for the `key`.

Reply:
in the firestore, I am storing this ``tags`` array in every doc, and referring to an id would be tough. I am looking for a way around though. Thank you for the review.